uvicorn/config.py

```diff
@@ -226,6 +226,7 @@ def __init__(
         headers: list[tuple[str, str]] | None = None,
         factory: bool = False,
         h11_max_incomplete_event_size: int | None = None,
+        before_graceful_exit_hook: Callable[[], Any] | str | None = None,
```

Comment:
Might as well make it async right?

Reply:
@abersheeran can you clarify that in the type hint? I realize that Any encompasses Awaitable but being explicit would be better.

Reply:
> @abersheeran can you clarify that in the type hint? I realize that Any encompasses Awaitable but being explicit would be better.

Okay.